Cargo.toml

```diff
@@ -9,10 +9,12 @@ repository = "https://github.com/spruceid/siwe-recap/"
 
 [dependencies]
 base64 = "0.12"
-iri-string = "0.6"
+iri-string = { version = "0.6", features = ["serde"] }
 siwe = "0.5"
 serde_json = { version = "1" }
 serde = { version = "1", features = ["derive"] }
 serde_with = { version = "2" }
 thiserror = "1"
 cid = "0.10"
+ucan-capabilities-object = "0.1"
+serde_jcs = "0.1"
```

Comment:
the EIP ordering rules turned out to be exactly those used in JCS, so I used it here to enforce them, otherwise writing ordered serialisation would have been quite tough for deeply nested objects (`BTreeMap<Ability, _>` would order correctly, but using `serde_json::Value` with `serde_json` for the caveat objects would not enforce the ordering)

Reply:
serde has a tendency to ignore duplicate fields. could you add a test to ensure we fulfill this requirement:
>  Objects in the att field (including nested objects) MUST NOT contain duplicate keys 

(i.e. deserialization should fail)

Reply:
this turns out to be pretty tricky, I will add the test into `ucan_capability_object` as we have a custom `Deserialize` impl for the nota benes where we can potentially enforce this

Reply:
Should I add an issue there?

Reply:
I'll add one now, looks like [we can use this at least for the first level of the map](https://docs.rs/serde_with/latest/serde_with/rust/maps_duplicate_key_is_error/index.html)